cmd/snap-confine/snap-confine.c

```diff
@@ -574,8 +575,11 @@ static void enter_non_classic_execution_environment(sc_invocation * inv,
 
 	/** Populate and join the device control group. */
 	struct snappy_udev udev_s;
-	if (snappy_udev_init(inv->security_tag, &udev_s) == 0)
-		setup_devices_cgroup(inv->security_tag, &udev_s);
+	if (snappy_udev_init(inv->security_tag, &udev_s) == 0) {
+		if (!sc_cgroup_is_v2()) {
```

Comment:
This will produce a generic cgroups v2 not supported warning. Perhaps we should inform which feature is actually getting disabled.

Reply:
I'm conflicted on this. There is no point in udev tagging if there isn't a device cgroup, but on the other hand, not doing it is a larger runtime change. With this proposed change, snap-device-helper will bail early since `/sys/fs/cgroup/devices/snap.*` should not exist (only snap-confine creates it thus far) so we *should* be fine. I suspect a number of spread tests will need to omit cgroup v2 systems.